src/Client.php

```diff
@@ -266,16 +273,34 @@ public function __call(string $endpoint, array $params = [])
             );
         }
 
+
+        if ( property_exists( $response_object, 'totalResultsCount' ) ) {
+            $this->lastTotalResultsCount = $response_object->totalResultsCount;
+        }
+
         // return the value of the root property from the response object (if the endpoint supports it)
         $root_property = $this->endpoints[$endpoint];
         if ($root_property !== false && property_exists($response_object, $root_property)
         ) {
+            if ( $this->lastTotalResultsCount === null && is_array($response_object->{$root_property})) {
+                $this->lastTotalResultsCount = count( $response_object->{$root_property} );
+            }
             return $response_object->{$root_property};
         }
 
+        if ( $this->lastTotalResultsCount === null && is_array($response_object)) {
```

Comment:
Looks like there is some duplication that can be generalized...

```php
if ( this->lastTotalResultsCount === null) {
  if ( $rootProperyExists ) {
    // case A
  } else {
    // case B
  }
}
```

Reply:
Well, I have just extended the existing logic which used a return statement in the if-clause. Hence, the fallback mechanism had to be implemented twice.

I have now simplified the logic by removing the first return statement and instead point the $response_object to it's child $root_property.